### DISCUSSION TITLE
Prevent scroll to top with NuxtLink

### DISCUSSION THREAD

**Discussion by madsh93** [nuxt/nuxt#25867] [Questions]
Hi!

I have a simple NuxtLink. This link only changes the query, therefore I don't need a scroll to top. 
How do I change this effect?

```js
        <NuxtLink
          :to="{
            query: {
              some_query: 'test'
            },
            replace: true,
          }"
        />

**Comment by owen2jan** [comment]
Are you trying to filter a list of items on same page? If so, you can try this approach :

1. Have an initial method to fetch items using query from the route.
2. Have another method for fetching and updating new data when filters change or on paginate etc.
3. update route after each fetch ( so that data persists even after reload ).

**Reply by madsh93** [reply]
Hi Owen

I am not looking for a workaround. But I was hoping I was missing something obvious.

Similar to something like Inertias:


![image](https://github.com/nuxt/nuxt/assets/8860723/27cc4547-03f3-4ab0-9956-d57048f0e89e)

I think it should be solvable by setting a custom scroll behavior.

**Reply by xjccc** [reply]
```
definePageMeta({
  scrollToTop: false
})
```

**Reply by owen2jan** [reply]
I'm assuming since you are "only changing the query" it is a **same page redirect**.

If I'm not wrong I think nuxt uses vue-router behind the scenes and **same page redirects don't update the page** unless you use a navigation guard like [beforeRouteEnter ](https://router.vuejs.org/guide/advanced/navigation-guards.html#In-Component-Guards) . Another workaround would be to have multiple redirects coming back to same page which is inefficient . You can also use [force](https://router.vuejs.org/api/interfaces/RouteLocationOptions.html#force) which triggers the navigation even if the location is the same as the current one although I'm not sure how to do this with the `<NuxtLink/>` component.

What @xjccc suggested will work although this behavior will persist even when redirecting from different page which can become a problem later. 

**nb:** I think forcing same page redirect isn't efficient. You would be much better fetching the needed data using `useFetch()` and updating the route afterwards( this will also maintain the current scroll ).

if you just want to change the query string and not update the page you can use `useRouter()` composable :
```
const route = useRouter()
route.push(route.path + "?some_query=test");
```

**Reply by madsh93** [reply]
Weirdly enough I copied the default router options:

https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/pages/runtime/router.options.ts

and added it to `app/router.options.ts` and it works. Not even sure why, I should have same settings by default, no?
Using router.push would be a great solution otherwise.

**Reply by owen2jan** [reply]
I haven't tried custom routing but using nuxt as is after installation, same page redirects didn't update the page by default. Same thing when using vue and vue-router.

**Reply by xjccc** [reply]
> I haven't tried custom routing but using nuxt as is after installation, same page redirects didn't update the page by default. Same thing when using vue and vue-router.

You can set props page-key at NuxtPage

**Reply by owen2jan** [reply]
> You can set props page-key at NuxtPage

this works!
